templates/addons/env/addons.parameters.yml

```diff
@@ -1,2 +1,6 @@
 Parameters:
+  EnvironmentSecurityGroup: !Ref EnvironmentSecurityGroup
```

Comment:
@lgarvey This parameter is currently only needed if one or more of the following conditions is satisfied...

* We need a `redis-cluster`
* We need a `shared` `aurora-postgres`
* We need an `rds-postgres`
* We need an `opensearch`

In the `aurora-postgres` template I made including the parameter optional so the linting is not upset by the unused parameter.

However, now when I try to deploy `update-supply-chain-information` using newly generated CloudFormation templates it goes...

```
➜ copilot env deploy --name development
✘ upload artifacts for environment development: parse environment addons: template does not require the parameter "EnvironmentSecurityGroup" in parameters file
```

I am a Jinja2 newbie, but I think it should be possible to only include it in this file if the conditions listed are satisfied?

I do suspect that we do not need this for version `0.1` though, we could just document deleting unused entries from addons.parameters.yml` and ticket this up for `0.2`.

What do you think?

Reply:
I definitely think it should at least be deferred to s separate pull request as this one already solves some major future developer headaches. I will create the ticket now, we can prioritise it later.

Reply:
Yes, this makes sense.  It should be fairy trivial to add this.